interop/xds/client/client.go

```diff
@@ -280,26 +377,32 @@ func makeOneRPC(c testpb.TestServiceClient, cfg *rpcConfig) (*peer.Peer, *rpcInf
 	return &p, &info, err
 }
 
-func sendRPCs(clients []testpb.TestServiceClient, cfgs []*rpcConfig, ticker *time.Ticker) {
+func sendRPCs(clients []testpb.TestServiceClient, ticker *time.Ticker) {
 	var i int
 	for range ticker.C {
-		go func(i int) {
-			// Get and increment request ID, and save a list of watchers that
-			// are interested in this RPC.
-			mu.Lock()
-			savedRequestID := currentRequestID
-			currentRequestID++
-			savedWatchers := []*statsWatcher{}
-			for key, value := range watchers {
-				if key.startID <= savedRequestID && savedRequestID < key.endID {
-					savedWatchers = append(savedWatchers, value)
-				}
+		// Get and increment request ID, and save a list of watchers that are
+		// interested in this RPC.
+		mu.Lock()
+		savedRequestID := currentRequestID
+		currentRequestID++
+		savedWatchers := []*statsWatcher{}
+		for key, value := range watchers {
+			if key.startID <= savedRequestID && savedRequestID < key.endID {
+				savedWatchers = append(savedWatchers, value)
 			}
-			mu.Unlock()
+		}
+		mu.Unlock()
+
+		// Get the RPC metadata configurations from the Configure RPC.
+		rpcCfgsMu.Lock()
+		cfgs := make([]*rpcConfig, len(rpcCfgs))
+		copy(cfgs, rpcCfgs)
+		rpcCfgsMu.Unlock()
 
-			c := clients[i]
+		for _, cfg := range cfgs {
+			go func(i int, cfg *rpcConfig) {
+				c := clients[i]
```

Comment:
Move this to line 401 and don't capture `i` in the closure.

Reply:
Made this change.